server/src/test/java/io/crate/planner/operators/OuterJoinRewriteTest.java

```diff
@@ -121,4 +122,59 @@ public void testFilterOnFullOuterJoinIsPartiallyPushedDownToTheRightSide() {
             """;
         assertThat(plan).isEqualTo(expectedPlan);
     }
+
+    @Test
+    public void test_filter_on_aliased_multiple_join_is_partially_pushed_down() {
```

Comment:
I think it is a lot easier to test single rule applications without the interaction of multiple rules if possible.